src/lib/reasoners/sat_solver_sig.ml

```diff
@@ -31,18 +31,25 @@
 (* We put an ml file for the module type, to avoid issues when
    building the lib *)
 
+type timeout_reason =
+  | Assume
+  | ProofSearch
+  | ModelGen
+
+type unknown_reason =
+  | Incomplete
+  | Memout
+  | Timeout of timeout_reason
+
+(* TODO: use an ADT and add Timeout when library starts acting like a
+   library. *)
```

Comment:
I think this comment can go away?

Reply:
@Stevendeo I can merge as soon as the comment has been removed ;)